core/controllers/profile.py

```diff
@@ -133,6 +132,16 @@ def get(self):
         })
         self.render_json(self.values)
 
+class ProfilePictureHandlerByUsername(base.BaseHandler):
+    """ Provides the dataURI of the profile picture of the specified user,
+    or none if no user picture is uploaded for the user with that ID."""
```

Comment:
If the username does not exist, I suggest surfacing a 404 error. You can do this by raising a self.PageNotFoundException.